myapp/main.py

```diff
@@ -0,0 +1,490 @@
+        db_session.delete(d)
+    db_session.commit()
+    return "Database has been cleaned"
+
+
+# 2.Common Side
+# common homepage(temporary)
+@app.route('/')
+def main():
+    return render_template('common_home.html')
+
+
+# 3.Student Side
+# student homepage
+@app.route('/student_home')
+def student_home():
+    return render_template('student_home.html')
+
+
+# A student can select among all current available labs
+@app.route('/student_enter_data')
+def student_enter_data():
+    lab_list = []
+    db_session = db.get_session()
+    query_all_lab = db_session.query(schema.Lab_info).all()
+    for lab in query_all_lab:
+        lab_list.append({'lab_name': lab.lab_name,
+                         'class_name': lab.class_name,
+                         'prof_name': lab.prof_name})
+    return render_template('student_enter_data.html',
+                           lab_list=lab_list)
+
+
+# A student can input data here
+@app.route('/student_data_entry/<lab_id>')
+def student_data_entry(lab_id):
+    rows_info = []
+    lab_info = []
+    db_session = db.get_session()
+    query_lab_info = db_session.query(schema.Lab_info).filter(
+        schema.Lab_info.lab_id == lab_id)
+    query_rows_info = db_session.query(schema.Lab_rows).filter(
+        schema.Lab_rows.lab_id == lab_id).order_by(schema.Lab_rows.row_order)
+
+    # Convert the objects to dictionaries to make them JSON serializable
+    for row in query_rows_info:
+        rows_info.append({'row_name': row.row_name, 'row_desc': row.row_desc,
+                          'value_type': row.value_type,
+                          'value_range': row.value_range,
+                          'value_candidates': row.value_candidates})
+    for lab in query_lab_info:
+        lab_info.append({'lab_id': lab.lab_id,
+                         'lab_name': lab.lab_name,
+                         'lab_desc': lab.lab_desc})
+    return render_template('student_data_entry.html',
+                           rows_info=rows_info, lab_info=lab_info)
+
+
+# store incoming data into the database
+@app.route('/_student_receive_data', methods=['POST'])
+def _student_receive_data():
+    jsonData = request.get_json()
+    lab_data = jsonData['lab_data']
+    db_session = db.get_session()
+    for d in lab_data:
+        student_name = d['student_name']
+        for r in d['rows_info']:
+            lab_id = r['lab_id']
+            row_name = r['row_name']
+            row_id = Generate_row_id(lab_id, row_name)
+            data_id = Generate_data_id(row_id, student_name)
+            db_session.add(schema.Lab_data(row_id=row_id, data_id=data_id,
+                                           student_name=student_name,
+                                           row_data=r['row_data']))
+    db_session.commit()
+    # return success to the ajax call from flask
+    return jsonify(success=True, data=jsonData)
+
+
+# 4.Admin Side
+# admin homepage
+@app.route('/admin_home')
```

Comment:
We will add a login page later after we know Reed login api and grab the username to identify their identity and thus redirect them to different home pages. The user names will be checked before they come to a homepage.